sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherClientBuilder.java

```diff
@@ -0,0 +1,273 @@
+            httpPipelinePolicies.add((context, next) -> {
+                context.getHttpRequest().getHeaders().put(AEG_SAS_TOKEN, sasToken.getSas());
+                return next.process();
+            });
+        } else {
+            httpPipelinePolicies.add(new AzureKeyCredentialPolicy(AEG_SAS_KEY, keyCredential));
+        }
+
+        httpPipelinePolicies.addAll(policies);
+
+        HttpPolicyProviders.addAfterRetryPolicies(httpPipelinePolicies);
+
+        httpPipelinePolicies.add(new HttpLoggingPolicy(httpLogOptions));
+
+        HttpPipeline buildPipeline = new HttpPipelineBuilder()
+            .httpClient(httpClient)
+            .policies(httpPipelinePolicies.toArray(new HttpPipelinePolicy[0]))
+            .build();
+
+
+        return new EventGridPublisherAsyncClient(buildPipeline, hostname, buildSerializer);
+    }
+
+    /**
+     * Build a publisher client with synchronous publishing methods and the current settings. Endpoint and a credential
+     * must be set (either keyCredential or sharedAccessSignatureCredential), all other settings have defaults and/or are optional.
+     * Note that currently the asynchronous client created by the method above is the recommended version for higher
+     * performance, as the synchronous client simply blocks on the same asynchronous calls.
+     * @return a publisher client with synchronous publishing methods.
+     */
+    public EventGridPublisherClient buildClient() {
+        return new EventGridPublisherClient(buildAsyncClient());
+    }
+
+    /**
+     * Add a policy to the current pipeline.
+     * @param httpPipelinePolicy the policy to add.
+     *
+     * @return the builder itself.
+     */
+    public EventGridPublisherClientBuilder addPolicy(HttpPipelinePolicy httpPipelinePolicy) {
+        this.policies.add(Objects.requireNonNull(httpPipelinePolicy));
+        return this;
+    }
+
+    /**
+     * Add a custom retry policy to the pipeline. The default is {@link RetryPolicy#RetryPolicy()}
+     * @param retryPolicy the retry policy to add.
+     *
+     * @return the builder itself.
+     */
+    public EventGridPublisherClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = retryPolicy;
+        return this;
+    }
+
+    /**
+     * Set the configuration of HTTP and Azure values. A default is already set.
+     * @param configuration the configuration to use.
+     *
+     * @return the builder itself.
+     */
+    public EventGridPublisherClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Set the domain or topic authentication using a key obtained from Azure CLI, Azure portal, or the ARM SDKs.
+     * @param credential the key credential to use to authorize the publisher client.
+     *
+     * @return the builder itself.
+     */
+    public EventGridPublisherClientBuilder keyCredential(AzureKeyCredential credential) {
```

Comment:
The method should be named `credential` to be consistent with other SDK builders.